content/api/vision/marker-ids.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Marker IDs
+---
+
+Each marker used in the game has an ID that corresponds to the marker type. You can use this within your code to determine what sort of marker you're looking at and make decisions based off of it.
+
+There are some useful constants which can be imported to do this. For example:
+
+```python
+from robot import WALL, TOKEN_ZONE_1, COLUMN_N_FACING_N
+```
+
+To check your marker matches one of these types, you can use the `in` operator:
+
+```python
+marker = r.camera.see()[0]  # Only the closest marker
```

Comment:
Could we introduce the tuple unpacking spelling of `marker, * = r.camera.see()` rather than indexing?

Reply:
Does that work for lists too?

I'm not sure I like attempting to introduce students to this _advanced_ syntax like this, as they may not understand what it's doing

Reply:
It does indeed work for lists. (It turns out that you do need to name the `*` variable: `x, *y = [...]`, though that's unrelated to the unpacking -- the unpacking works for lists in Python 2).

I think my main objections here are:
- list indexing in python is generally a signal that you're doing something a bit wrong
- we're encouraging code which is intolerant of error cases
- we're reinforcing the notion (which as noted previously I don't like) that the markers are sorted

Rethinking this: since we're not defining the `r` here explicitly, I'm not actually sure why we feel a need to define the marker explicitly either. It could be viable to skip these lines entirely.

If we're merely trying to introduce the `in` operator here, then maybe we should back up to something where it's clearer what the operator is doing:
``` python
num = 4
num in [3,1,4,1,5]
>>> True
fibonacci = [1,1,2,3,5]
num in fibonacci
>>> False
```

Reply:
There's a note on the api intro that states `r` is implicitly defined as a `Robot` instance, http://docs.sourcebots.co.uk/api/#setup.

I don't think it's necessary to define _what_ the `in` operator does (the whirlwind tour should handle that), but I think keeping it to known, simple syntax like that is totally fine here. 

I'm not exactly sure what the issue is with indexing is, however maybe using `.pop()` is more explicit?

Reply:
Essentially indexing usually doesn't match up with what you're trying to do and there are frequently  better python idioms to use that would match up.

Normally you'd want to loop over all the markers to find the one you're looking for, so you wouldn't index into the list. I'd rather not encourage indexing as it leads to this sort of thing:
``` python
for i in range(len(markers)):
    marker = markers[i]
```

I'm not sure how to avoid some form of element access here, hence the suggestion to just avoid it. (`.pop()` doesn't avoid that and does introduce issues like modifying the list).